### PR TITLE
Use internal variables to generate proper tuf-reposerver url in start.sh:get_credentials function

### PR DESCRIPTION
**Pull request by ddunlap2005**
The tuf-reposerver url fix is not dependent on the macOS changes. Not sure if I am creating this pull request correctly.

The two commits needed are:

- Fix tuf-reposerver url when generating server keys
- Resolve HTTP 423 Locked error

The second fix is a bit crude. If there is an actual entity we could wait on, that would be better. I did try a 2s sleep, but that was not sufficient.

### CODE REVIEW

File: scripts/start.sh [hunk 1]
```diff
@@ -254,7 +259,9 @@ get_credentials() {
 
   http_2xx_or_4xx --ignore-stdin --check-status POST "${reposerver}/api/v1/user_repo" "${namespace}"
 
-  id=$(http --ignore-stdin --check-status --print=h GET http://tuf-reposerver.ota.local/api/v1/user_repo/root.json x-ats-namespace:default | grep x-ats-tuf-repo-id | awk '{print $2}' | tr -d '\r')
+  sleep 5s
+  id=$(http --ignore-stdin --check-status --print=h GET "${reposerver}/api/v1/user_repo/root.json" "${namespace}" | grep -i x-ats-tuf-repo-id | awk '{print $2}' | tr -d '\r')
+  echo "id=${id}"
```
Comment (patrickvacek) [inline scripts/start.sh:264]: I think echoing the `id` here is unnecessary

File: scripts/start.sh [hunk 2]
```diff
@@ -254,7 +259,9 @@ get_credentials() {
 
   http_2xx_or_4xx --ignore-stdin --check-status POST "${reposerver}/api/v1/user_repo" "${namespace}"
 
-  id=$(http --ignore-stdin --check-status --print=h GET http://tuf-reposerver.ota.local/api/v1/user_repo/root.json x-ats-namespace:default | grep x-ats-tuf-repo-id | awk '{print $2}' | tr -d '\r')
+  sleep 5s
```
Comment (patrickvacek) [inline scripts/start.sh:262]: I still get the 423, but retrying works fine for me. I'd prefer a better way to wait for the resource to become available (we do something like that in the aktualizr tests) but for now I don't mind this.

**Review by patrickvacek — COMMENTED** [review]
Appears to work for me, but I use Linux, so it might be good to get a Mac user to verify. @houcros / @jochenschneider / @tkfu can you take a quick look?

> The tuf-reposerver url fix is not dependent on the macOS changes.

Agreed, that I can confirm since I had to use a similar fix.

> Not sure if I am creating this pull request correctly.

Looks fine to me. It includes the commits in https://github.com/advancedtelematic/ota-community-edition/pull/113 but that doesn't bother me.

**Comment by patrickvacek** [comment]
If you remove the `echo`, I'll merge this.

**Comment by ddunlap2005** [comment]
The echo is now removed.

**Review by patrickvacek — APPROVED** [review]
Thanks! I'm calling it good.

**Pull request merged**